infra/periodic_killer.tf

```diff
@@ -55,23 +55,34 @@ set -euxo pipefail
 apt-get update
 apt-get install -y jq
 
+echo "$(date -Is -u) boot" > /root/log
+
 cat <<CRON > /root/periodic-kill.sh
 #!/usr/bin/env bash
 set -euo pipefail
+echo "\$(date -Is -u) start"
 
 PREFIX=vsts-
 MACHINES=\$(/snap/bin/gcloud compute instances list --format=json | jq -c '.[] | select(.name | startswith("'\$PREFIX'")) | [.name, .zone]')
 
 for m in \$MACHINES; do
-    /snap/bin/gcloud -q compute instances delete \$(echo \$m | jq -r '.[0]') --zone=\$(echo \$m | jq -r '.[1]')
+    MACHINE_NAME=\$(echo \$m | jq -r '.[0]')
+    MACHINE_ZONE=\$(echo \$m | jq -r '.[1]')
+    # We do not want to abort the script on error here because failing to
+    # reboot one machine should not prevent trying to reboot the others.
+    /snap/bin/gcloud -q compute instances delete \$MACHINE_NAME --zone=\$MACHINE_ZONE || true
 done
+
+echo "\$(date -Is -u) end"
```

Comment:
Same here why are we escaping this?

Reply:
Answering my own question based on the one above, we are also in the context of a `cat`.